# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/TEJAS7356/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All


I chose the Refresh / Content Opportunity Scoring lane because I want to identify which pages should be prioritized for review based on their available performance and content signals. The goal is to use the dataset to find pages that may need refreshing, improving, protecting, or monitoring, and create a useful way to prioritize them.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

The decision is which pages should be prioritized for review and potential content improvement. A content or SEO team could use the ranking to decide which pages to refresh, expand, protect, or monitor first. A wrong recommendation could cause the team to spend time and resources on a page that does not need attention, while a page that actually needs improvement could be overlooked.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [5]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

df = pd.read_csv(
    "/content/flyrank-ml-internship/data/raw/content_refresh_anonymized.csv"
)

df.head()

print("Number of pages:", len(df))
print("Median impressions in 90 days:", df["impressions_90d"].median())
print("Median CTR:", df["ctr"].median())


Number of pages: 30000
Median impressions in 90 days: 731.0
Median CTR: 0.07


The starter dataset contains 30,000 pages. The median page has 731 impressions over 90 days, while the median CTR is 0.07. These numbers show that the dataset contains a large number of pages with measurable visibility and click performance, making it useful to explore how pages can be prioritized for review and potential content improvement.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


This work will provide observed and directional insights from the available dataset and can support decisions about which pages may deserve review. It can identify patterns and prioritize pages based on measurable signals. However, it will not prove that any specific factor causes a page's performance, and it will not claim to predict or explain Google's ranking algorithm.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.